blueflood-elasticsearch/src/test/java/com/rackspacecloud/blueflood/io/ElasticsearchRestHelperTest.java

```diff
@@ -0,0 +1,87 @@
+package com.rackspacecloud.blueflood.io;
+
+import com.rackspacecloud.blueflood.utils.Metrics;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import static com.rackspacecloud.blueflood.utils.Metrics.counter;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*", "javax.net.ssl.*"})
+@PrepareForTest(fullyQualifiedNames = "org.apache.http.impl.client.*")
+public class ElasticsearchRestHelperTest {
+
+  private ElasticsearchRestHelper helper;
+  private HttpClientBuilder mockBuilder = mock(HttpClientBuilder.class);
+  private CloseableHttpClient mockHttpClient = mock(CloseableHttpClient.class);
+  private CloseableHttpResponse mockHttpResponse = mock(CloseableHttpResponse.class);
+  private StatusLine statusLine = mock(StatusLine.class);
+
+  @Before
+  public void setup() throws IOException {
+    // HttpClient dependency is nested inside pool manager build logic.
+    // Mocking Http response requires mocking multiple instances of Apache http client resources.
+    mockStatic(HttpClients.class);
+    when(HttpClients.custom()).thenReturn(mockBuilder);
+    when(mockBuilder.setConnectionManager(any())).thenReturn(mockBuilder);
+    when(mockBuilder.build()).thenReturn(mockHttpClient);
+    when(mockHttpResponse.getStatusLine()).thenReturn(statusLine);
+
+    resetMetricRegistry();
+    helper = ElasticsearchRestHelper.getInstance();
```

Comment:
The `getInstance` method will hopefully change to returning a static singleton in the future. I created the `getConfigurableInstance` method for unit tests, where you always want to obtain a fresh instance, in case you need to change the app config and get a new instance of the rest helper.